modules/data.land/tests/testthat/test-clip_and_move_raster_file.R

```diff
@@ -0,0 +1,57 @@
+library(testthat)
+library(terra)
+library(sf)
+
+# load function under test
+source("../../clip_and_move_rasters.R")  # adjust relative path as needed
+
+# helper to create a small test raster
+make_raster <- function(crs = "EPSG:4326") {
+  r <- terra::rast(matrix(1:16, 4, 4),
+                   extent = terra::ext(0, 4, 0, 4), crs = crs)
+  f <- tempfile(fileext = ".tif")
+  terra::writeRaster(r, f, filetype = "GTiff", overwrite = TRUE)
+  f
+}
+
+test_that("clip & mask works: output clipped to polygon bbox and masked", {
+  in_r <- make_raster()
+  on.exit(unlink(in_r), add = TRUE)
+  # box polygon (1,1)-(3,3)
+  poly <- sf::st_as_sfc(sf::st_bbox(c(xmin=1, ymin=1, xmax=3, ymax=3)), crs = "EPSG:4326")
+  out_f <- tempfile(fileext = ".tif")
+  clip_and_move_raster_files(in_r, poly, out_f, mask = TRUE)
+  expect_true(file.exists(out_f))
+  r_out <- terra::rast(out_f)
+  # extent == polygon bbox
+  expect_equal(terra::ext(r_out), terra::ext(sf::st_bbox(poly)))
+  # some values NA (corners) and some not (center)
+  vals <- terra::values(r_out)
+  expect_true(any(is.na(vals)))
+  expect_true(any(!is.na(vals)))
+  unlink(out_f)
```

Comment:
This would likely be easier to write cleanly with `withr::with_tempfile()`, but on.exit is OK here, just use it consistently please
```suggestion
  out_f <- tempfile(fileext = ".tif")
  clip_and_move_raster_files(in_r, poly, out_f, mask = TRUE)
  expect_true(file.exists(out_f))
  r_out <- terra::rast(out_f)
  on.exit(unlink(out_f), add = TRUE)
  # extent == polygon bbox
  expect_equal(terra::ext(r_out), terra::ext(sf::st_bbox(poly)))
  # some values NA (corners) and some not (center)
  vals <- terra::values(r_out)
  expect_true(any(is.na(vals)))
  expect_true(any(!is.na(vals)))
```

Reply:
...Thanks for the confusing diff, GitHub. My suggested change is just to move the `unlink(out_f)` up and wrap it inside an `on.exit`.

Reply:
changed to use withr